module-system/module-implementations/examples/sov-accessory-state/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+#[cfg(feature = "native")]
+pub mod query;
+
+use sov_modules_api::{CallResponse, Context, Error, Module, ModuleInfo};
+use sov_state::codec::BorshCodec;
+use sov_state::storage::{StorageKey, StorageValue};
+use sov_state::{Prefix, StateValue, WorkingSet};
+
+#[derive(ModuleInfo)]
+pub struct AccessorySetter<C: sov_modules_api::Context> {
+    #[address]
+    pub address: C::Address,
+    #[state]
+    pub latest_value: StateValue<String>,
+}
+
+#[derive(borsh::BorshDeserialize, borsh::BorshSerialize, Debug, PartialEq)]
+pub enum CallMessage {
+    SetValue(String),
+    SetValueAccessory(String),
+}
+
+impl<C: Context> AccessorySetter<C> {
+    /// Sets a value in the JMT state.
+    fn set_value(&self, s: String, working_set: &mut WorkingSet<C::Storage>) {
+        self.latest_value.set(&s, working_set);
+    }
+
+    /// Sets a value in the non-JMT state.
+    fn set_value_accessory(&self, s: String, working_set: &mut WorkingSet<C::Storage>) {
+        let prefix = Prefix::new(self.address.as_ref().to_vec());
```

Comment:
It would be grat to hide all this `prefix` handling from a user. Have you considered adding another field type in the module? For example we have `StateValue<String>` and we could add `NotAuthenticatedValue<String>` or something like that. Then would generate `prefix` the same way as for the StateXX fields. 
@preston-evans98 any thoughts about the API?

Reply:
Yes, adding non-JMT versions of `StateValue`, `StateMap` (and possibly `StateVec`) would be simple. I'm considering it, but I wanted to get some feedback first. The main drawback would be code duplication. We'd effectively almost copy-paste much of that code, which I'd like to avoid. The main alternative I see is to add a flag to the state containers we have already, and depending on the value of the flag we set/get/delete from JMT data or from accessory data. WDYT?

Reply:
Dumping some ideas:
Another approach would be to add 2 type level flags:
`struct Authenticated`
`struct NotAuthenticated`
Then `StateXX` takes it as a generic argument with default value `Authenticated` but we can opt out with 
```
value = StateValue<u32, SomeCodec, NotAuthenticated>
```
Maybe it is too complex tough.

Reply:
I'm unsure what I prefer myself. Both approaches have pros and cons.

Reply:
I implemented some `AccessoryState*` state containers, I'll resolve this conversation and we can address any comments about those separately :+1: